tests/unittest/_torch/visual_gen/test_control_kernels.py

```
@@ -0,0 +1,233 @@
+        with pytest.raises(AssertionError, match="requires a CUDA tensor, got device=cpu"):
+            bilateral_filter(cpu, 9, 75.0, 75.0)
+        with pytest.raises(AssertionError, match="requires a CUDA tensor, got device=cpu"):
+            resize_linear_u8(cpu, 4, 4)
+        with pytest.raises(AssertionError, match="requires a CUDA tensor, got device=cpu"):
+            canny_edges(torch.zeros(3, 1, 8, 8, dtype=torch.uint8), 100, 200)
+
+    def test_rejects_non_uint8(self):
+        # Silently casting would be a data copy on the inference path; the
+        # kernels require the caller to hand over the dtype they expect.
+        f32 = torch.zeros(1, 8, 8, 3, dtype=torch.float32, device="cuda")
+        with pytest.raises(AssertionError, match="requires uint8 frames, got dtype=torch.float32"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)